.github/workflows/sbom_generation.yaml

```diff
@@ -0,0 +1,33 @@
+name: Generate SBOM
+on: [workflow_dispatch]
+
+jobs:
+  install-trivy:
+    name: Install Trivy
+    runs-on: ubuntu-latest
+    env:
+      TRIVY_VERSION: "0.55.2"
+      TRIVY_ARCH: ${{'64bit'}}
+      TRIVY_SHA256: ${{'09b8730cf16d63b65f1e282f63dfaa16183a143a045b51467864f5e634025a5f' }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
why is this necessary?

Reply:
From the [Git checkout action repo](https://github.com/actions/checkout), 

> Only a single commit is fetched by default, for the ref/SHA that triggered the workflow. Set fetch-depth: 0 to fetch all history for all branches and tags. Refer [here](https://docs.github.com/actions/using-workflows/events-that-trigger-workflows) to learn which commit $GITHUB_SHA points to for different events.

I thought that this was important.